### ISSUE TITLE
False positive no-else-return

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pylint#2295]
### Steps to reproduce
1. Put this code in `test.py`:
```python
def f(x, y, z):
    if x:
        return x
    elif y:
        return y
    return z
```

2. Run `pylint --disable=C test.py`.

### Current behavior
```
test.py:2:4: R1705: Unnecessary "else" after "return" (no-else-return)
```

### Expected behavior
No complaints about `else`, because there is no `else` in this code.

### pylint --version output
```
pylint 2.0.0
astroid 2.0.0.dev4
Python 3.5.3 (default, Jan 19 2017, 14:11:04)
[GCC 6.3.0 20170118]
```

**Comment by RemiCardona** [comment]
yeah, got that one too, it lit up our CI.

**Comment by PCManticore** [comment]
This one is simple. `pylint` expects you to drop the `elif` clause resulting in:
```
if x:
   return x
if y:
   return y
return z
```
The text of the message could be better to suggest either `elif` or `else` depending on the actual clause.
